src/osemgrep/targeting/Unit_glob.ml

```diff
@@ -0,0 +1,65 @@
+(*
+   Unit tests for glob pattern parsing and matching.
+*)
+
+open Printf
+
+let test pattern path matches () =
+  let pat = Glob_lexer.parse_string pattern in
+  let compiled_pat =
+    Glob_matcher.compile ~source:(Glob_matcher.string_loc pattern) pat
+  in
+  let res = Glob_matcher.run compiled_pat path in
+  printf
+    "pattern: %s\n\
+     path: %s\n\
+     expected match: %B\n\
+     actual match: %B\n\
+     pattern info:\n\
+     %s\n\
+     %s\n"
+    pattern path matches res
+    (Glob_matcher.show_pattern pat)
+    (Glob_matcher.show compiled_pat);
+  Alcotest.(check bool) "equal" matches res
+
+let tests =
+  Testutil.pack_tests "Glob"
+    [
+      ("simple", test "abc" "abc" true);
+      ("absolute path", test "/a/b" "/a/b" true);
+      ("star 1", test "*.c" "hello.c" true);
+      ("star 2", test "*.c" "hello.h" false);
+      ("question 1", test "a.?" "a.c" true);
+      ("question 2", test "a.?" "b.c" false);
+      ("relative path 1", test "ab/*.c" "ab/hello.c" true);
+      ("relative path 2", test "ab/*.c" "hello.c" false);
+      ("relative path 3", test "ab/*.c" "ab/" false);
+      ("relative path 4", test "ab/*.c" "x/ab/hello.c" false);
```

Comment:
so this means the pattern is kinda "anchored"? ab/*.c does not match x/ab/hello.c"

Reply:
yes, the glob patterns are anchored on both ends. I added that to the mli. The patterns found in gitignore/semgrepignore files are rewritten to use this correctly.

Say my project contains:
```
.git/
thing.c
src/
src/.gitignore
src/hello.c
src/tmp/
src/tmp/hello2.c
```
and `src/.gitignore` contains:
```
*.c
```
`*.c` is expanded into `/src/**/*.c` so as to match `/src/hello.c` and `/src/tmp/hello2.c`.